relay-common/src/utils.rs

```diff
@@ -6,45 +6,110 @@ use regex::Regex;
 
 use crate::macros::impl_str_serde;
 
-/// A simple glob matcher.
-///
-/// Supported are `?` for a single char, `*` for all but a slash and
-/// `**` to match with slashes.
-#[derive(Clone)]
-pub struct Glob {
-    value: String,
-    pattern: Regex,
+/// Glob options represent the underlying regex emulating the globs.
+#[derive(Debug)]
+struct GlobPatternOpts<'g> {
```

Comment:
nit: I would like to give this struct a more descriptive name but I don't have one. Maybe something like `GlobPatternGroups` or `GlobPatternBuildingBlocks`?